### PR TITLE
Add credo and dialyzer

### PR DESCRIPTION
**Pull request by dantswain**
Two changes to the default credo config:
1. Set max line length to 100
2. Disable nested alias check

### CODE REVIEW

**Comment by dantswain** [comment]
I don't feel super strongly about having dialyzer run as part of the travis build.  It shouldn't take very long after the first build, though.  We'll have a slow build once we merge to master but then any branches should inherit the master cache and it shouldn't be bad unless we change Elixir or OTP versions.

**Comment by dantswain** [comment]
@cjonesy you might also want to sign up for ebert and add it to this repo: https://ebertapp.io/

**Comment by cjonesy** [comment]
@dantswain ebert has been added now.

**Comment by dantswain** [comment]
Sweet.  With ebert enabled we don't need credo running in travis, so I'll remove that.

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        # For others you can set parameters
+
+        # If you don't want the `setup` and `test` macro calls in ExUnit tests
+        # or the `schema` macro in Ecto schemas to trigger DuplicatedCode, just
+        # set the `excluded_macros` parameter to `[:schema, :setup, :test]`.
+        #
+        {Credo.Check.Design.DuplicatedCode, excluded_macros: []},
+
+        # You can also customize the exit_status of each check.
+        # If you don't want TODO comments to cause `mix credo` to fail, just
```
[the remark this answers is not in the document]
Comment (dantswain) [reply .credo.exs]: Thanks, Ebert, for finding the TODO in the credo config file where you can configure if credo detects TODOs. :|

[the remark this answers is not in the document]
Comment (cjonesy) [reply .credo.exs]: TODO: Shut up,  Ebert.

**Pull request merged**